### ISSUE TITLE
Improve YouTube support 

### ISSUE THREAD

**Issue by The0x539** [wallabag/wallabag#2149]
- [ ] Fetch video descriptions, channel, and other non-volatile metadata
- [ ] Use video length for reading time
- [ ] Make the 'original link' / 'original article' buttons lead to something other than an XML document, either the full-page video embed or the actual youtube.com/watch page
- [ ] [Don't display thumbnail as large header above actual video embed](https://cloud.githubusercontent.com/assets/7350294/16135565/fdf1c77a-33e9-11e6-9406-41ef51e87cce.png)

**Comment by Kaligule** [comment]
We could use youtube-dl for that. It is pretty good at providing these information for a wide range of video websites.

**Comment by tcitworld** [comment]
Unfortunately youtube-dl is python-only therefore this feature will not be available on many instances.

**Comment by jlnostr** [comment]
Because youtube-dl can return the metadata as JSON, maybe a public youtube-dl instance could be installed on the wallabag server, so the self-hosted instances can communicate with it to fetch the metadata. In the advanced configuration this could be enabled/disabled.

**Comment by j0k3r** [comment]
We don't extra software to extract metadata from Youtube, the oembed endpoint is enough:

https://www.youtube.com/oembed?format=json&url=https://www.youtube.com/watch?v=uhbuEUoxntQ

```json
{
    "provider_name": "YouTube",
    "width": 480,
    "thumbnail_width": 480,
    "thumbnail_url": "https://i.ytimg.com/vi/uhbuEUoxntQ/hqdefault.jpg",
    "author_name": "RT",
    "html": "<iframe width=\"480\" height=\"270\" src=\"https://www.youtube.com/embed/uhbuEUoxntQ?feature=oembed\" frameborder=\"0\" allowfullscreen></iframe>",
    "height": 270,
    "author_url": "https://www.youtube.com/user/RussiaToday",
    "version": "1.0",
    "thumbnail_height": 360,
    "type": "video",
    "provider_url": "https://www.youtube.com/",
    "title": "Clash of epochs: Drone speared at history festival in central Russia"
}
```

**Comment by The0x539** [comment]
What of video length?

**Comment by j0k3r** [comment]
Well, why not using the Youtube API then?

**Comment by Kaligule** [comment]
While there is nothing wrong with using the youtube Api in principle,
what would we do about all the other video websits people might bag?

**Comment by j0k3r** [comment]
We'll need a dedicated class to handle each of them.
Or we can use an external service to retrieve information about them (I'm less for this solution).

**Comment by Kaligule** [comment]
Btw, what is the problem with youtube-dl being written in python? If it
is installed, can't we just call it? And if not, we could return a blind
test that says so.

**Comment by tcitworld** [comment]
> Btw, what is the problem with youtube-dl being written in python? If it is installed, can't we just call it? And if not, we could return a blind test that says so.

We can have several levels of getting information : 
* oembed
* youtube-dl if installed
* our own instance that can deliver information from youtube-dl
* our own service that can deliver information from the YouTube API.

I am quite opposed that wallabag itself uses the YouTube API.

**Comment by lapineige** [comment]
@tcitworld +1 for avoiding Youtube API

I would say that just using 1) and have a proper video integration* would already be a massive improvement (instead of blank article with no title / article without the video).
Other video providers are likely to demand another special adaptation indeed, but youtube is used a lot more - that means for the users a great first step.
For others providers (like dailymotion or video), it would need another solution - but what I mean is that improving at least youtube support is important.

@tcitworld you mean it would not be available on hosting without python support, right ?
If we don't find a workaround to find the duration of the video, should wallabag use embed information by default, and use youtube-dl is possible ?
Or add an option (desactivated by default) to use Youtube API ?

*using the youtube embeded frame address / converting a youtube video address to the embeded one: basically changing `youtube.com/watch?v=` to `youtube.com/embed/
If we bag a youtube video directly (and not an article containing an embeded video) wallabag could also retrieve the description of the video and add it.

**Comment by aaa2000** [comment]
Maybe, the https://github.com/essence/essence library can be used

**Comment by tcitworld** [comment]
@aaa2000 Yeah, there was some discussions about it on https://github.com/wallabag/wallabag/issues/880

**Comment by thomasfrivold** [comment]
You should go with youtube-dl. It is an impressive project by itself by now.

As a user unwound be super happy and do not care at all about my python requirement.

I cannot  think of anywhere outside of shared webhosts that a python dependency cannoy be easily met.

**Comment by wu-lee** [comment]
I also have this problem.  That the youtube links always have to be manually unpacked is a major pain for me.  Whereas knowing the length of the video is barely important. I suggest the latter shouldn't impede fixing the former. +1 for using simply using the oembed data.

**Comment by N07070** [comment]
Does anyone have any update on this ?

**Comment by ungeschneuer** [comment]
+1

**Comment by bcutter** [comment]
https://github.com/wallabag/wallabag/issues/6567#issuecomment-2993692488 + https://github.com/wallabag/wallabag/issues/3638

Still fails to open an original YT video URL as every app out there uses the URL and only Wallabag offers the source URL (visible on the right here):

![Image](https://github.com/user-attachments/assets/d6c316c3-808d-4e7a-adcd-77356a119f6c)

Very very frustrating.

**Comment by thejohnha** [comment]
Hi, just started testing wallabag and the very first page I saved looks not so nice. It would be great to have a placeholder image and link it to the youtube video. This is the page I tested: https://thetabletopfamily.com/easy-pen-and-paper-games/
